design-documents/error-reporting/reduce-q-ty-of-error-reports.md

```diff
@@ -0,0 +1,82 @@
+
+**For example:**
+
+***The error response to user:***
+
+![The error response to user](./imgs/example-error-report.png "The error response to user")
+
+***The report file:***
+
+![Example the error report file](./imgs/directory-list.png "Example the error report file")
+
+***The message in the `<magento_root>/var/log/exception.log`:***
+
+![Example the exception messages in the exception log file](./imgs/exception-mesages.png "Example the exception messages in the exception log file")
+
+There is only one report file, although there are several messages in the exception log. Messages are the same but happened at different times
+
+***Report File Content***
+
+![Example the exception messages in the error report](./imgs/error-report.png "Example the exception messages in the error report")
+
+Of course, the decision to save only unique reports will not solve the problem of accumulating report files in a directory, this decision only delay the problem for another time. Anyway, over time, the number of files can grow up to several hundred thousand and working with such a directory will again be difficult.
+
+To solve this problem, we can save reports in subdirectories with a configurable level of nesting.
+For example, to add a new property `dir_nesting_level` to `<magento_root>/pub/errors/local.xml` for configuring directory nesting level to saving reports. And additionally, we can use the environment variable.
+
+***Variants of name:***
+
+1. MAGE_LOG_EXCEPTION_REPORT_DIR_NESTING_LEVEL
+1. MAGE_EXCEPTION_REPORT_DIR_NESTING_LEVEL
+1. MAGE_LOG_REPORT_DIR_NESTING_LEVEL
+1. MAGE_REPORT_DIR_NESTING_LEVEL
+1. REPORT_DIR_NESTING_LEVEL
+
+
+The default value of the env variable/property is 0, in this case, report files will be saved to the directory `magento_root/var/report/<hash>`, without nesting.
+If the environment variable has value more than zero, then subdirectories will be created. Each subdirectory will have a name like first two symbols of hash with offset on two symbols for next sub level.
+
+***For Example:***
+
+If we have hash sha256('')= 44ffb1087a44e61b018b3cdee72284d017f22e52755c24e5c85cbac1647ae7a7
+
+1. MAGE_REPORT_DIR_NESTING_LEVEL=0 -> <magento_root>/var/report/44ffb1087a44e61b018b3cdee72284d017f22e52755c24e5c85cbac1647ae7a7
+1. MAGE_REPORT_DIR_NESTING_LEVEL=1 -> <magento_root>/var/report/**44**/44ffb1087a44e61b018b3cdee72284d017f22e52755c24e5c85cbac1647ae7a7
+1. MAGE_REPORT_DIR_NESTING_LEVEL=2 -> <magento_root>/var/report/**44**/**ff**/44ffb1087a44e61b018b3cdee72284d017f22e52755c24e5c85cbac1647ae7a7`
+....
+1. MAGE_REPORT_DIR_NESTING_LEVEL=32 -> <magento_root>/var/report/**44**/**ff**/**b1**/**08**/**7a**/**44**/**e6**/**1b**/**01**/**8b**/**3c**/**de**/**e7**/**22**/**84**/**d0**/**17**/**f2**/**2e**/**52**/**75**/**5c**/**24**/**e5**/**c8**/**5c**/**ba**/**c1**/**64**/**7a**/**e7**/**a7**/44ffb1087a44e61b018b3cdee72284d017f22e52755c24e5c85cbac1647ae7a7
```

Comment:
We need to come up with recommended nesting level or some formula for calculating optimal nesting level. It should be published in docs.